src/pages/Home/home.js

```diff
@@ -1,7 +1,156 @@
+    }
+  };
+
+  componentDidMount() {
+    this.getBookData("horror");
+    this.getBookData("romance");
+    this.getBookData("mystery");
+    this.getBookData("nonfiction");
+    this.getBookData("history");
+  }
+
+  MappedHorrorData = () => {
+    const mappedData = this.state.horror.map((book) => {
+      return (
+        <Card>
+          <Link to={`/book/${book.volumeInfo.industryIdentifiers[0].identifier}`}>
+            <div>{book.volumeInfo.title}</div>
+            <div>
+              <img
+                src={`${book.volumeInfo.imageLinks.thumbnail}`}
+                alt={`${book.volumeInfo.title}`}
+              />
+            </div>
+          </Link>
+        </Card>
+      );
+    });
+    return <Container>{mappedData}</Container>;
+  };
+
+  MappedRomanceData = () => {
+    const mappedData = this.state.romance.map((book) => {
+      return (
+        <Card>
+          <Link to={`/book/${book.id}`}>
+            <div>{book.volumeInfo.title}</div>
+            <div>
+              <img
+                src={`${book.volumeInfo.imageLinks.thumbnail}`}
+                alt={`${book.volumeInfo.title}`}
+              />
+            </div>
+          </Link>
+        </Card>
+      );
+    });
+    return <Container>{mappedData}</Container>;
+  };
+
+  MappedMysteryData = () => {
+    const mappedData = this.state.mystery.map((book) => {
+      return (
+        <Card>
+          <Link to={`/book/${book.id}`}>
+            <div>{book.volumeInfo.title}</div>
+            <div>
+              <img
+                src={`${book.volumeInfo.imageLinks.thumbnail}`}
+                alt={`${book.volumeInfo.title}`}
+              />
+            </div>
+          </Link>
+        </Card>
+      );
+    });
+    return <Container>{mappedData}</Container>;
+  };
+
+  MappedNonfictionData = () => {
+    const mappedData = this.state.nonfiction.map((book) => {
+      return (
+        <Card>
+          <Link to={`/book/${book.id}`}>
+            <div>{book.volumeInfo.title}</div>
+            <div>
+              <img
+                src={`${book.volumeInfo.imageLinks.thumbnail}`}
+                alt={`${book.volumeInfo.title}`}
+              />
+            </div>
+          </Link>
+        </Card>
+      );
+    });
+    return <Container>{mappedData}</Container>;
+  };
+
+  MappedHistoryData = () => {
+    const mappedData = this.state.history.map((book) => {
+      return (
+        <Card>
+          <Link to={`/book/${book.id}`}>
+            <div>{book.volumeInfo.title}</div>
+            <div>
+              <img
+                src={`${book.volumeInfo.imageLinks.thumbnail}`}
+                alt={`${book.volumeInfo.title}`}
+              />
+            </div>
+          </Link>
+        </Card>
+      );
+    });
+    return <Container>{mappedData}</Container>;
+  };
+
+  render() {
+    return (
+      <>
+        <h1>Horror</h1>
+        {this.state.horror.length && <div>{this.MappedHorrorData()}</div>}
```

Comment:
I still think that, if you initialise the state to null and you just check this.state.genre instead of this.state.genre.length would be better.  You can try it out and see if it works

Reply:
Yup...you are absolutely right. Thanks for pointing it out!